lib/bike_brigade_web/live/itinerary_live/index.ex

```diff
@@ -0,0 +1,54 @@
+defmodule BikeBrigadeWeb.ItineraryLive.Index do
+  use BikeBrigadeWeb, :live_view
+
+  alias BikeBrigade.LocalizedDateTime
+  alias BikeBrigade.Riders
+  alias BikeBrigade.Delivery
+
+  import BikeBrigadeWeb.CampaignHelpers
+
+  @impl true
+  def mount(_params, _session, socket) do
+    today = LocalizedDateTime.today()
+
+    {:ok,
+     socket
+     |> assign(:page, :itinerary)
+     |> assign(:page_title, "Itinerary")
+     |> assign(:today, today)
+     |> assign(:deliveries, fetch_deliveries(socket, today))}
+  end
+
+  @impl true
+  def handle_params(_params, _url, socket) do
+    {:noreply, socket}
+  end
+
+  defp fetch_deliveries(socket, today) do
+    Riders.list_campaigns_with_task_counts(
+      Riders.get_rider!(
+        if not is_nil(socket.assigns.current_user.rider_id),
+          do: socket.assigns.current_user.rider_id,
+          else: 26
+      ),
+      # today
+      ~D[2022-06-27]
+    )
+  end
+
+  defp get_location(assigns) do
```

Comment:
I like that you named this function `get_location` since it is in fact getting something from the database, it's good to have functions give you that kind of clue from the name.

This also is a clue that we want to split up the concerns - when you have a " function component" -  a function that's returning HTML, we want to keep it [pure](https://en.wikipedia.org/wiki/Pure_function) - that is just rendering the input.

I think what you want to do is something like `@delivery.location.address` or `@campaign.location` if you rename as suggested above.

If you try that now you'll see that the location is missing - it will an association that isn't loaded. We can use `Repo.preload` on the campaign inside of `fetch_deliveries`, but better yet would be to update `Riders.list_campaigns_with_task_counts` to include that preload